Sources/Basic/Path.swift

```diff
@@ -118,12 +124,31 @@ public struct AbsolutePath {
         return AbsolutePath(self, subpath)
     }
     
-    /// Returns the absolute path with the contents of the string (interpreted
-    /// as a relative path, not a single path component) appended.
-    public func appending(_ str: String) -> AbsolutePath {
-        return AbsolutePath(self, str)
+    /// Returns the absolute path with an additional literal component appended.
+    ///
+    /// This method should only be used in cases where the input is guaranteed
+    /// to be a valid path component (i.e., it cannot be empty, contain a path
+    /// separator, or be a pseudo-path like '.' or '..').
+    public func appending(component name: String) -> AbsolutePath {
+        assert(name != "" && name != "." && name != ".." && !name.contains("/"))
```

Comment:
Cool.  Because I think the next step will be that anything that uses the `appending(component:)` method will want to preflight the component string and throw an error if, say, a module name ends up consisting of just a `.` character or whatever.